### PR TITLE
Correctly pass app_name to all email templates.

### PR DESCRIPTION
**Pull request by erikjohnston**
We didn't do this for e.g. registration emails.

### CODE REVIEW

File: synapse/push/mailer.py [hunk 1]
```diff
@@ -291,6 +291,13 @@ async def send_email(self, email_address, subject, template_vars):
         if raw_to == "":
             raise RuntimeError("Invalid 'to' address")
 
+        template_vars = {
+            "app_name": self.app_name,
+            "server_name": self.hs.config.server.server_name,
+        }
```
Comment (clokep) [inline synapse/push/mailer.py:297]: Were these passed manually in some cases that we should now remove from the callers?

Comment (erikjohnston) [reply synapse/push/mailer.py:297]: Goodpointwellmade.

File: changelog.d/7829.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix bug where we did not always pass in `app_name` or `server_name` to email templates, including e.g. for registration emails.
```
Comment (clokep) [inline changelog.d/7829.bugfix:1]: Has this "always" been this way or was it broken at some point? (Do we need to add a "broken since v..."?)

Comment (erikjohnston) [reply changelog.d/7829.bugfix:1]: I *think* this has always been broken for these emails, as they were added later at different times

**Pull request merged**